### PR TITLE
Adding a basic Meta arithmetic implementation 

### PR DESCRIPTION
**Pull request by tharanga**
This will get binary incr/decr compatibility for the ascii protocol

### CODE REVIEW

File: evcache-core/src/main/java/com/netflix/evcache/operation/EVCacheAsciiOperationFactory.java [hunk 1]
```diff
@@ -23,5 +28,9 @@ public ExecCmdOperation execCmd(String cmd, ExecCmdOperation.Callback cb) {
         return new ExecCmdOperationImpl(cmd, cb);
       }
 
+    public MutatorOperation mutate(Mutator m, String key, long by, long def,
```
Comment (arunagrawal84) [inline evcache-core/src/main/java/com/netflix/evcache/operation/EVCacheAsciiOperationFactory.java:31]: nit: would prefer descriptive names like incrementAmount, defaultValue, expiry, opCallback.

File: evcache-core/src/main/java/net/spy/memcached/protocol/ascii/MetaArithmeticOperationImpl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import net.spy.memcached.ops.StatusCode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+
+public class MetaArithmeticOperationImpl extends EVCacheOperationImpl implements
```
Comment (sumanth-pasupuleti) [inline evcache-core/src/main/java/net/spy/memcached/protocol/ascii/MetaArithmeticOperationImpl.java:18]: nit: good to include description for public classes

File: evcache-core/src/main/java/net/spy/memcached/protocol/ascii/MetaArithmeticOperationImpl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        mutator = m;
+        key = k;
+        amount = amt;
+        def = defaultVal;
+        exp = expiry;
+        readingValue = false;
+    }
+
+    @Override
+    public void handleLine(String line) {
```
Comment (sumanth-pasupuleti) [inline evcache-core/src/main/java/net/spy/memcached/protocol/ascii/MetaArithmeticOperationImpl.java:59]: given that core of the logic in the class is protocol implementation, I am not reviewing the logic itself.

**Review by sumanth-pasupuleti — APPROVED** [review]

File: evcache-core/src/main/java/com/netflix/evcache/operation/EVCacheAsciiOperationFactory.java (hunk 1 above)
Comment (tharanga) [reply evcache-core/src/main/java/com/netflix/evcache/operation/EVCacheAsciiOperationFactory.java:31]: I kept the nomenclature similar to other op implementations, such as MutatorOperationImpl.

**Pull request merged**